src/chatbot/src/chatbot/aiml/DefaultSubs.py

```diff
@@ -48,8 +50,8 @@
     "hers": "his",
     "herself": "himself",
 }
-
-defaultPerson = {
+# These now match the comment above
+defaultPerson2 = {

```

Comment:
Indeed this is what is happening i.e. implementation is opposite to the
spec at http://www.alicebot.org/documentation/aiml-reference.html#person

So the question is how widely are the person and person2 tags used in the
base aiml correctly or incorrectly, and am I responding to a rare
correct/incorrect
use of the tag that didn't get edited to match the semantics.  Answer:  in
the hr_aiml (original set used when I started working) no person2 tags
appear
328 person tags appear.   Spot checking, most appear to be oriented toward
giving a third person answer about a third subject, which wouldn't seem to
require a substitution.

I know that I've always seen pronoun substitution bugs, and spent a day or
two grabbing alternative alice aiml sets to see whether we weren't missing
some substitution reduction aiml that's supposed to handle that.  At some
point I discovered the sub table, but only now noticed the inconsistency
between the variable names and comments.

In one such aiml set we have called 'standard' person appears 188 times,
 person2 once in some kind of test, non-operational context.

Given all this, I guess the right thing is to make pyAIML match the spec
consistently and try to track down bad templates that really want I to you
or me
transformations but are giving I to him now.   The spec says that person2
is hard to use because you-> I  and you->me are difficult to handle.

Maybe the 328 vs. 188 explains why I see a lot of bad subsitutions -
perhaps somebody wrote more aiml based on the incorrect implementation.

I'm not sure there is any course of action here that results in immediate
good behavior in all cases,  since the aiml sets seem to be full of
guessing
what was used in a * match.

I think the model of an ALICE conversation includes a lot more talking
about third parties,  most conversations with Sophia tend to be about her,
the listener, humans and general, and crafted answers to special third
parties like David Hanson.





On Sat, Aug 20, 2016 at 10:46 AM, Linas Vepštas <notifications@github.com>
wrote:

> In src/chatbot/src/chatbot/aiml/DefaultSubs.py
> <https://github.com/hansonrobotics/HEAD/pull/168#discussion_r75580623>:
>
> > @@ -48,8 +50,8 @@
> >      "hers": "his",
> >      "herself": "himself",
> >  }
> > -
> > -defaultPerson = {
> > +# These now match the comment above
> > +defaultPerson2 = {
>
> this looks incorrect. According to the AIML spec, person2 converts from
> 1st person to second person for example, I <--> you while the below is
> clearly for 3rd person ...
>
> Likewise, the person rule is supposed to convert 1st to 3rd person.
>
> I'm guessing that the "I like bannanas" rule is broken -- it is using
> person instead of person2 -- I suggest that the correct fix is to fix the
> rule, instead of the definition of the conversion.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/hansonrobotics/HEAD/pull/168/files/85ee952bbb4ef504d4b96fb53f07abcfdd7700a7#r75580623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF3yl3nX7e2T412jplF8KwmF32Hpg9Ubks5qhyFlgaJpZM4Jo5ec>
> .
>